FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.m

```diff
@@ -201,6 +218,14 @@ - (instancetype)initWithGoogleAppID:(NSString *)appID
   });
 }
 
+- (BOOL)areInstallationRegistrationParametersEqualToCurrent:
+    (FIRInstallationsStoredRegistrationParameters *)parameters {
+  NSString *APIKey = self.APIService.APIKey;
+  NSString *projectID = self.APIService.projectID;
+  return (parameters.APIKey == APIKey || [parameters.APIKey isEqual:APIKey]) &&
+         (parameters.projectID == projectID || [parameters.projectID isEqual:projectID]);
```

Comment:
parameters.APIKey == APIKey
Is this for comparing the address to be the same as well?

Reply:
I use `parameters.APIKey == APIKey` to cover the case when they are both `nil`. `[parameters.APIKey isEqual:APIKey]` returns `NO` in this case.  It is the easiest way to cover all equality cases I came up with, but I am open for suggestions.

Reply:
nice! I didn't know that.

Reply:
yeah, I didn't pay attention to it until the integration tests failures. Basically if e.g. `parameters.APIKey == nil` then any message to it will return `nil` which is casted to `NO`.